### PR TITLE
117 url encode session

### LINKED ISSUE #117: URL encoding needed for _session request
The `POST` to `_session` for the cookie auth in 2.0 needs to URL encode the username and password fields.

### PR DESCRIPTION
**Pull request by ricellis**
*What*
Fixes #117 - encode the username and password of the `_session` request.

*How*
Apply `x-www-form-urlencoded` encoding to the user and password fields.

*Testing*
Added new test to check encoding.

reviewer @emlaver 
reviewer @rhyshort

### CODE REVIEW

File: src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java [hunk 1]
```diff
@@ -145,15 +160,14 @@ private String getCookie(URL url){
...
-        //Map<String,Object> jsonResponse = gson.fromJson(new InputStreamReader(responseStream));
-        JsonObject jsonResponse = gson.fromJson(new InputStreamReader(responseStream), JsonObject.class);
+        JsonObject jsonResponse = gson.fromJson(new InputStreamReader(responseStream), JsonObject
+                .class);
 
         // only check for ok:true, https://issues.apache.org/jira/browse/COUCHDB-1356
         // means we cannot check that the name returned is the one we sent.
-       //return jsonResponse.containsKey("ok") && (Boolean) jsonResponse.get("ok");
         return jsonResponse != null && jsonResponse.has("ok")
                 && !jsonResponse.get("ok").isJsonNull();
```
Comment (rhyshort) [inline src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java]: This doesn't check that `ok` has the value `true`, just that it exists, and it isn't `null`

Comment (ricellis) [inline src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java]: Fixed in  3f7c798.

**Comment by rhyshort** [comment]
+1

**Comment by emlaver** [comment]
+1

**Pull request merged**